src/nomad_measurements/xrd/readers.py

```diff
@@ -175,67 +175,59 @@ def read_rigaku_rasx(file_path: str, logger: BoundLogger=None) -> Dict[str, Any]
         Dict[str, Any]: The X-ray diffraction data in a Python dictionary.
     '''
     reader = RASXfile(file_path, verbose=False)
-    data = reader.data
-    metainfo = reader.meta
-    p_data = reader.get_RSM()
+    scan_info = reader.get_scan_info()
+    p_data = reader.get_1d_scan(logger)
+    source = reader.get_source_info()
 
-    if not data.shape[0] == 1:
-        if logger is not None:
-            logger.warning(
-                '2D scan currently not supported. '
-                'Taking the data from the first line scan.'
-            )
-        for key, data in p_data.items():
-            if isinstance(data, np.ndarray) and data.ndim == 2:
-                p_data[key] = data[0,:].squeeze()
+    count_time = None
+    scan_axis = None
 
-        if not len(np.unique(p_data['Omega'])) == 1:
-            raise ValueError(
-                'Unexpected array of Omega angles. Should contain same value of angles.'
-            )
-        p_data['Omega'] = p_data['Omega'][0]
-    metainfo = metainfo[0]
+    if scan_info:
+        required_keys = ['Mode','SpeedUnit','PositionUnit','Step', 'Speed']
+        if all(key in scan_info and scan_info[key] for key in required_keys):
+            if scan_info['Mode'].lower() == 'continuous':
+                speed_unit = ureg(scan_info['SpeedUnit'].replace('min','minute'))
+                count_time_unit = ureg(scan_info['PositionUnit']) / speed_unit
+                count_time = (
+                    np.array([scan_info['Step'] / scan_info['Speed']])
+                    * count_time_unit
+                )
 
-    count_time = None
-    if metainfo['ScanInformation']['Mode'].lower() == 'continuous':
-        speed_unit = ureg(
-            metainfo['ScanInformation']['SpeedUnit'].replace('min','minute')
-        )
-        count_time_unit = ureg(metainfo['ScanInformation']['PositionUnit']) / speed_unit
-        count_time = (
-            metainfo['ScanInformation']['Step']
-            / metainfo['ScanInformation']['Speed']
-        )
-        count_time = np.array([count_time]) * count_time_unit
+        scan_axis = scan_info.get('AxisName',None)
 
-    axis = {}
-    for ax in ['Omega', 'Chi', 'Phi']:
-        if not isinstance(p_data[ax], np.ndarray):
-            axis[ax] = np.array([p_data[ax]])
-        else:
-            axis[ax] = p_data[ax]
-    source = metainfo['HardwareConfig']['xraygenerator']
+    def set_quantity(quantity_list):

```

Comment:
I see the point. But `quantity_list` could be either an empty list (in case the key is not present, the defaultdict in get methods of RASXFile will return []) or a list containing 2 indices. The current way can handle both. I think unpacking approach will lead to different number of arguments in the two cases. 

I should add this information about quantity_list in the docstring.

Reply:
I would then give the arguments a default value instead.
Something like:
```python
def set_quantity(value: Any=None, unit: str=None) -> Any:
    if unit is None:
       return value
    return value * ureg(unit)
```

Reply:
Look good! I have modified the function in the way you suggested. But I used a slightly different if condition, that recognizes empty unit string as missing unit. 
```
def set_quantity(value: Any=None, unit: str=None) -> Any:
    if not unit:
       return value
    return value * ureg(unit)
```